src/crawlers/cheerio_crawler.js

```diff
@@ -412,7 +413,12 @@ class CheerioCrawler extends BasicCrawler {
         this.prepareRequestFunction = prepareRequestFunction;
         this.postResponseFunction = postResponseFunction;
         this.proxyConfiguration = proxyConfiguration;
-        this.persistCookiesPerSession = persistCookiesPerSession;
+
+        if (this.useSessionPool) {
+            this.persistCookiesPerSession = persistCookiesPerSession !== undefined ? persistCookiesPerSession : true;
```

Comment:
same here

Reply:
And it's done in exact same way here - if `persistCookiesPerSession` is not `undefined` - use the actual value, if it is `undefined` - set it to `true` when session pool is used, and set it to `false` when not. It's never `undefined`, or am I missing something?